core/src/GenericGF.h

```diff
@@ -45,33 +47,46 @@ class GenericGF
 	static const GenericGF& AztecData8();
 	static const GenericGF& MaxiCodeField64();
 
-	GenericGFPoly zero() const {
-		return _zero;
+	/**
+	* @return the monomial representing coefficient * x^degree
+	*/
+	GenericGFPoly& setMonomial(GenericGFPoly& poly, int degree, int coefficient) const
+	{
+		assert(degree >= 0);
+
+		if (coefficient == 0)
+			degree = 0;
+
+		poly._field = this;
+		poly._coefficients.resize(degree + 1);
+		std::fill(poly._coefficients.begin(), poly._coefficients.end(), 0);
+		poly._coefficients.front() = coefficient;
+
+		return poly;
 	}
 
-	GenericGFPoly one() const {
-		return _one;
+	GenericGFPoly& setZero(GenericGFPoly& poly) const {
+		return setMonomial(poly, 0, 0);
 	}
 
-	/**
-	* @return the monomial representing coefficient * x^degree
-	*/
-	GenericGFPoly buildMonomial(int degree, int coefficient) const;
+	GenericGFPoly& setOne(GenericGFPoly& poly) const {
+		return setMonomial(poly, 0, 1);
+	}
 
 	/**
 	* Implements both addition and subtraction -- they are the same in GF(size).
 	*
 	* @return sum/difference of a and b
 	*/
-	static int AddOrSubtract(int a, int b) {
+	int addOrSubtract(int a, int b) const {
 		return a ^ b;
 	}
 
 	/**
 	* @return 2 to the power of a in GF(size)
 	*/
 	int exp(int a) const {
-		return _expTable.at(a);
+		return _expTable[a];
```

Comment:
I removed it mostly because it seemed inconsistent to check the `_expTable` but no the `_logTable` access. A quick performance comparison showed that the extra check is irrelevant -> I reverted this line.